### ISSUE TITLE
V4 - Plugin Custom Resolution Issues

### ISSUE THREAD

**Issue by ac360** [serverless/serverless#12508]
When referencing a Plugin in `serverless.yml` outside of the Service, users are having trouble resolving them if 1) they are referencing a file (e.g. .js) or 2) the plugin might not feature a package.json with a `main` property.

https://github.com/serverless/serverless/issues/12503#issuecomment-2123569664
https://github.com/serverless/serverless/issues/12503#issuecomment-2123850006
https://github.com/serverless/serverless/issues/12503#issuecomment-2123940195

**Comment by ac360** [comment]
> Custom plugins reference when using V3
> plugins:
>  ../infrastructure/queuer/serverless-fargate-plugin/dist/index.js
When using V4, I had to update, as it's now looking for a package.json. (doco suggests a JS file is fine)
> plugins:
>  ./../../infrastructure/queuer/serverless-fargate-plugin
Otherwise I get:
> no such file or directory, open '/codebuild/output/srcxxx/src/github.com/xxx/web/node_modules/serverless-lift/package.json'

@richard-stafflink Thanks for the additional info. In V4, we switched to ESM, which likely affected this behavior. Let's see if we can restore it.

**Comment by ac360** [comment]
For others who are experiencing this problem, please include:

• Your serverless.yml plugin references (we need to see the exact syntax).
• What your directory structure looks like.

**Comment by richard-stafflink** [comment]
Here is the difference in custom plugin resolution, there are not the same:
* v3: https://github.com/serverless/serverless/blob/v3/lib/classes/plugin-manager.js#L167
* v4: https://github.com/serverless/serverless/blob/main/lib/classes/plugin-manager.js#L308

Also, it would be handy if the Docs could be updated, as they are not inline with the current implementation.
V4 will use either a package.json file (with `main` property) or `plugin.js` (https://github.com/serverless/serverless/blob/main/lib/classes/plugin-manager.js#L292)

**Comment by ac360** [comment]
We're looking into not having the behavior change across versions ATM.

**Comment by kchecoBasis** [comment]
```
org: orgABC
app: app123
service: 'merchant'
frameworkVersion: '3'

package:
    individually: true
provider:
    name: aws
    runtime: nodejs18.x
    stage: ${opt:stage, 'dev'}
    region: us-east-1
    memorySize: 1024
    iam:
        role:
            statements: ${file(../../common.serverless.yml):statements}
    vpc: ${file(../../common.serverless.yml):vpc}
    apiGateway:
        restApiId:
            ...

    environment:
        ...

functions:
    ...

plugins:
    - serverless-bundle
    - serverless-offline

custom:
    bundle:
        forceExclude: ${file(../../common.serverless.yml):forceExclude}
        ignorePackages: ${file(../../common.serverless.yml):ignorePackages}
    cronEnabled:
        prod: true
```


The folder structure
```
    /
        /node_modules
        /services
            /service1
                ...
                serverless.yml
                package.json
            /service2
                ...
        common.serverless.yml
        package.json
````

**Comment by ac360** [comment]
We've added a few fixes into this in `4.0.26` which reinstates the loading behavior with v3. Closing unless this is still reported to be a problem

**Comment by nonso-caleb** [comment]
I am not certain if this is the right place. Unable to use serverless-appsync-plugin in v4. is there a fix for that
